src/App.tsx

```diff
@@ -15,48 +16,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortBy {
+  Name = 'name',
+  Length = 'length',
+  Default = '',
+}
+
+type SortType = SortBy | '';
+
+function sortGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+): string[] {
+  const copyOfGoods = [...goods];
+
+  switch (sortType) {
+    case SortBy.Name:
+      copyOfGoods.sort((good1, good2) => {
+        return good1.localeCompare(good2);
+      });
+      break;
+    case SortBy.Length:
+      copyOfGoods.sort((good1, good2) => {
+        return good1.length - good2.length;
+      });
+      break;
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    copyOfGoods.reverse();
+  }
+
+  return copyOfGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReversed] = useState(false);
+
+  const [sortType, setSortType] = useState<SortBy>(SortBy.Default);
+
+  const sortedGoods = sortGoods(goodsFromServer, sortType, isReversed);
+  const resetFilter = () => {
```

Comment:
```suggestion
  const sortedGoods = sortGoods(goodsFromServer, sortType, isReversed);

  const resetFilter = () => {
```

add please space between function and variable